examples/graph3d/16_styling_surface.html

```diff
@@ -0,0 +1,92 @@
+<!doctype html>
+<html>
+<head>
+  <title>Graph 3D | Styling Surface</title>
+
+  <style>
+    body {font: 10pt arial;}
+  </style>
+
+  <script type="text/javascript" src="../../dist/vis-graph3d.min.js"></script>
+
+  <script type="text/javascript">
+    var data = null;
+    var graph = null;
+
+    function custom(x, y) {
+      return (Math.sin(x/50) * Math.cos(y/50) * 50 + 50);
+    }
+
+    // Called when the Visualization API is loaded.
+    function drawVisualization() {
+      var style = document.getElementById('style').value;
+
+      // Create and populate a data table.
+      data = new vis.DataSet();
+      // create some nice looking data with sin/cos
+      var counter = 0;
+      var steps = 50;  // number of datapoints will be steps*steps
+      var axisMax = 314;
+      var axisStep = axisMax / steps;
+      for (var x = 0; x < axisMax; x+=axisStep) {
+        for (var y = 0; y < axisMax; y+=axisStep) {
+          var value = custom(x,y);
+          data.add({id:counter++,x:x,y:y,z:value,style:value});
+        }
+      }
+
+      // specify options
+      var options = {
+        width:  '600px',
+        height: '600px',
+        style: 'surface',
+        showPerspective: true,
+        showGrid: true,
+        showShadow: false,
+        keepAspectRatio: true,
+        verticalRatio: 0.5,
+      };
+
+      if (style === 'hue') {
+        options.surfaceColors = {
+          hue: {
+            start: 40,
+            end: 180,
+            saturation: 50,
+            brightness: 100,
+            colorStops: 8, // How many colour gradients do we want
+          }
+        };
+      } else if(style === 'html') {
+        options.surfaceColors = [
+          '#FFFFFF',
+          '#6A0DAD',
+          '#000000',
+        ];
+      } // else use defaults.
+
+      // Instantiate our graph object.
+      var container = document.getElementById('mygraph');
+      graph = new vis.Graph3d(container, data, options);
+
+      document.getElementById('style').onchange = drawVisualization;
+    }
+  </script>
+  
+</head>
+
+<body onload="drawVisualization();">
+<p>
+  <label for="style"> Style:
+    <select id="style">
+      <option value="default">Default</option>
+      <option value="hue">Hue</option>
+      <option value="html">HTML Hex</option>
```

Comment:
I would prefer if these were radio buttons at code blocks showing the configuration that is actually being supplied to the library but this is not a necessity.